logstash-core/lib/logstash/state_resolver.rb

```diff
@@ -25,12 +25,12 @@ def resolve(pipelines, pipeline_configs)
         end
       end
 
-      running_pipelines = pipeline_configs.collect(&:pipeline_id)
+      configured_pipelines = pipeline_configs.collect(&:pipeline_id)
 
       # If one of the running pipeline is not in the pipeline_configs, we assume that we need to
       # stop it.
-      pipelines.keys
-        .select { |pipeline_id| !running_pipelines.include?(pipeline_id) }
+      pipelines_registry.running_pipelines.keys
```

Comment:
I _think_ that your changes make better sense than the previous code, which had a variable called `running_pipelines` containing pipelines that _may-or-may-not_ be running.

Your code now checks to make sure that the _configured_ pipelines are _actually_ running before adding a stop action. If this is intentional, then :+1: